### PR TITLE
Add Support for signatureVersion v4. Can be set in Config Object.

### PR DESCRIPTION
**Pull request by slamus**
Hi,

I had aproblem access my bucket because AWS wouldn't allow me to access my bucket with a signature version different from v4.
I just added the possibility to set signature version in config, with default set at v4.

### CODE REVIEW

**Comment by slamus** [comment]
Hello,

Also Added option to filter listed keys with a string or regex via `String.prototype.match` js method

File: README.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ const lambda = new S3Lambda({
   secretAccessKey: 'aws-secret-key',   // Optional. (falls back on local AWS credentials)
   showProgress: true,                  // Optional. Show progress bar in stdout
   verbose: true,                       // Optional. Show all S3 operations in stdout (GET, PUT, DELETE)
+  signatureVersion: 'v4',                       // Optional. Signature Version used in Authentication. Defaults to "v4"
```
Comment (wellsjo) [inline README.md]: spacing issue here

File: README.md [hunk 2]
```diff
@@ -46,6 +47,7 @@ lambda.context({
   prefix: 'prefix/',         // The prefix of the files to use - s3-lambda will operate over every file with this prefix.
   marker: 'prefix/file1',    // Optional. Start at the first file with this prefix. If it is a full file path, starts with next file. Defaults to null.
   endPrefix: 'prefix/file3', // Optional. Process files up to (not including) this prefix. Defaults to null.
+  match: /2017/i, // Optional. Process files matching this regex / string. Defaults to null.
```
Comment (wellsjo) [inline README.md]: spacing

File: lib/S3Lambda.js [hunk 3]
```diff
@@ -96,6 +98,9 @@ class S3Lambda extends S3 {
               prefix: context.prefix,
               key
             }))
+            if (match) {
+              sources = sources.filter((line)=>line.key.match(match))
```
Comment (wellsjo) [inline lib/S3Lambda.js]: formatting (please use spaces before/after `=>`)

**Comment by wellsjo** [comment]
Thank you! This looks good, I just have a few minor nitpicks.

**Comment by slamus** [comment]
Is it good now ? :)

**Comment by wellsjo** [comment]
awesome, thanks!

**Pull request merged**